newIDE/app/src/ObjectEditor/Editors/Model3DEditor.js

```diff
@@ -260,32 +286,136 @@ const Model3DEditor = ({
         newModel3d => {
-          setModel3D(newModel3d);
+          setGltf(newModel3d);
         }
       );
     },
     [project]
   );
-  getModel3D(properties.get('modelResourceName').getValue());
+  getGltf(properties.get('modelResourceName').getValue());
+  const onChangeModelResourceName = React.useCallback(
+    (modelResourceName: string) => {
+      getGltf(modelResourceName);
+    },
+    [getGltf]
+  );
 
-  const onChangeProperty = React.useCallback(
-    (property: string, value: string) => {
-      objectConfiguration.updateProperty(property, value);
-      forceUpdate();
+  const model3D = React.useMemo<THREE.Object3D | null>(
+    () => {
+      if (!gltf) {
+        return null;
+      }
+      const clonedModel3D = SkeletonUtils.clone(gltf.scene);
+      const threeObject = new THREE.Group();
+      threeObject.rotation.order = 'ZYX';
+      threeObject.add(clonedModel3D);
+      return threeObject;
     },
-    [objectConfiguration, forceUpdate]
+    [gltf]
   );
 
-  const onChangeModelResourceName = React.useCallback(
-    (modelResourceName: string) => {
-      getModel3D(modelResourceName);
+  const [rotationX, setRotationX] = React.useState<number>(
+    parseFloat(properties.get('rotationX').getValue()) || 0
+  );
+  const [rotationY, setRotationY] = React.useState<number>(
+    parseFloat(properties.get('rotationY').getValue()) || 0
+  );
+  const [rotationZ, setRotationZ] = React.useState<number>(
+    parseFloat(properties.get('rotationZ').getValue()) || 0
+  );
+  const onRotationChange = React.useCallback(
+    () => {
+      setRotationX(parseFloat(properties.get('rotationX').getValue()));
+      setRotationY(parseFloat(properties.get('rotationY').getValue()));
+      setRotationZ(parseFloat(properties.get('rotationZ').getValue()));
+    },
+    [properties]
+  );
+  const modelSize = React.useMemo<{ x: number, y: number, z: number } | null>(
+    () => {
+      if (!model3D) {
+        return null;
+      }
+      // These formulas are also used in:
+      // - gdjs.Model3DRuntimeObject3DRenderer._updateDefaultTransformation
+      // - Model3DRendered2DInstance
+      model3D.rotation.set(
+        (rotationX * Math.PI) / 180,
+        (rotationY * Math.PI) / 180,
+        (rotationZ * Math.PI) / 180
+      );
+      model3D.updateMatrixWorld(true);
+      const boundingBox = new THREE.Box3().setFromObject(model3D);
+      return {
+        x: boundingBox.max.x - boundingBox.min.x,
+        y: boundingBox.max.y - boundingBox.min.y,
+        z: boundingBox.max.z - boundingBox.min.z,
+      };
     },
-    [getModel3D]
+    [model3D, rotationX, rotationY, rotationZ]
+  );
+
+  const [width, setWidth] = React.useState<number>(
+    parseFloat(properties.get('width').getValue()) || 0
+  );
+  const [height, setHeight] = React.useState<number>(
+    parseFloat(properties.get('height').getValue()) || 0
+  );
+  const [depth, setDepth] = React.useState<number>(
```

Comment:
Use `useState(() => ...)` to avoid useless calls to properties.get at each render (this can avoid multiple calls to wasm, always good)